CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+## Setup for Code Contributions
+
+### Prerequisites
+
+##### MacOS
+- XCode Command Line Tools `xcode-select --install`
+- If you already have the full Xcode installed, go to menu `Xcode -> Open Developer Tool -> More Developer Tools....`
+- Install Node.js >= 16
+
+##### Windows
+- Install current version of Python3
+- Install Visual Studio Code and add the Microsoft C/C++ extension
+- Install Node.js >= 16
+
+##### Linux
+
+- Install Python > 3.7
+- Install `make`, `gcc`, and `g++`
+- Install Node.js >= 16
+
+#### Repository Setup
+- Fork the project.
+- Open a Terminal and navigate to where you want the project repository to exist
+- Clone your forked repository
+    ```sh
+    git clone https://github.com/<your-github-user>/club_website.git
+    ```
+- Navigate into the root directory
+    ```sh
+    cd club_website
+    ```
+- Install required packages
+    ```sh
+    npm install 
+    ```
+- Switch to beta branch
+    ```sh
+    git checkout beta
+    ```
+- Start server
+    ```sh
+    npm run dev
+    ```
+- Set the upstream remote
+    ```sh
+    git remote add upstream https://github.com/NSC-Computer-Science-Club/club_website.git
+    ```
+- Verify
+    ```sh
+    git remote -v
+    ```
+
+#### Find Something to Work On
+Navigate to the `Issues` tab and use the filters to see open issues. Take note of the `good first issue` and `help wanted` labels.
+
+Remember to pull the most recent work from the upstream beta branch before starting your work!
+
+```sh
+git pull upstream beta
+```
+
+#### Submit a Pull Request
+
+- Create your feature branch
+    ```sh
+    git checkout -b <your initials>-<specific_feature>
```

Comment:
This is under their fork, so there's no need to include their name a second time. Also, it might be good to clarify how to name the specific feature.

Reply:
How do you want the feature to be named? I just used initials cause that's what we did at my last job. Do you have a naming convention you would like to see here? I just assumed people would name it after the feature but perhaps that is not a good idea :-)

Reply:
> How do you want the feature to be named? I just used initials cause that's what we did at my last job. Do you have a naming convention you would like to see here? I just assumed people would name it after the feature but perhaps that is not a good idea :-)

I think a good guideline is to keep it at one to two words, lowercase, and with hyphens or underscores instead of spaces (we can use one or the other, or both, up to you). I'm not sure how your job handled things, but imo initials make sense when everyone's working on the same repo and aren't using forks (or multiple people are on the same fork), so that duplicates can be avoided. With individual forks, every new branch is owned by the same person, so initials don't add too much value.